src/util/json_irep.cpp

```diff
@@ -102,9 +103,10 @@ irept json_irept::convert_from_json(const jsont &in) const
   for(const auto &keyval : in.object)
     have_keys.push_back(keyval.first);
   std::sort(have_keys.begin(), have_keys.end());
-  if(have_keys!=std::vector<std::string>{"comment", "id", "namedSub", "sub"})
-    throw "irep JSON representation is missing one of needed keys: "
-      "'id', 'sub', 'namedSub', 'comment'";
+  INVARIANT((have_keys==
```

Comment:
This is deserialisation related and should remain an exception (with a proper exception type)